### ISSUE TITLE
feat: ion-tabs support navigation animation

### ISSUE THREAD

**Issue by HyperLife1119** [ionic-team/ionic-framework#23653]
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already include this feature request, without success.

### Describe the Feature Request

let `ion-tabs` component support navigation animation, and supports routing navigation animation.

### Describe the Use Case

Allow developers to customize the animation when tab switching.
```html
<ion-tabs [animation]="animation"></ion-tabs>
```

### Describe Preferred Solution

Add an animation property to the `ion-tabs` component.

**Comment by HyperLife1119** [comment]
I think this feature is very important, it can help developers achieve similar Android native view pager switching dynamics. 
Hope to see its implementation in the v6 version. 
@liamdebeasi

**Comment by davideas** [comment]
I also agree to have an animation in this component: currently content just pop in from nowhere when tab is tapped having no clue if it comes from left or right., which makes double thinking: _Did a new page appear or I am still in same tabs page?_

**Comment by digaus** [comment]
Also interested in this feature. Would be nice if someone could look into it.

**Comment by tetsuwanadam** [comment]
Would love to be able to use this in my own apps as well.

**Comment by WhatsThatItsPat** [comment]
I would like a subtle version of this, which is perhaps different than the requests above.

Facebook has a subtle transition when switching tabs which comes in handy when you're viewing the same content in different tab stacks. In this GIF, we're looking at the Ionic page under both the home tab and the menu tab. The transition helps the user understand that the tabs have been changed. The direction of the transition depends on whether the new tab is to the right or left of the current tab.

![facebook-tab-transition](https://github.com/ionic-team/ionic-framework/assets/764290/ba76df72-898e-447a-ad8e-29abffc87ba5)

**Comment by singatias** [comment]
I'm not sure if this is related, but when using custom animations with <ion-router-outlet [animation]="customPageAnimation"> inside <ion-tabs>, the function never gets called. Adding a debugger; statement or console.log for debugging in my customPageAnimation function doesn't trigger. Brand new project using angular 18 and ionic 8.

**Comment by Trevor-143** [comment]
> I'm not sure if this is related, but when using custom animations with `<ion-router-outlet [animation]="customPageAnimation">` inside `<ion-tabs>`, the function never gets called. Adding a `debugger;` statement or `console.log` for debugging in my `customPageAnimation` function doesn't trigger. Brand new project using angular 18 and ionic 8.

I have just spent some time trying to make it work with no success.
